pkg/generator.go

```diff
@@ -1105,6 +1160,19 @@ func parseReplaceType(t string) *replaceType {
 		ret.alias = r[0]
 		t = r[1]
 	}
+
+	// Match type parameter substitution
+	match := regexp.MustCompile(`\[(.*?)\]$`).FindStringSubmatch(t)
+	if len(match) >= 2 {
+		if match[1][:1] == "-" {
```

Comment:
I was initially confused why you did that instead of:

```go
match[1][0]
```

but realized the range operator will return a string instead of a single byte. To make this cleaner, I'd instead do this:

```go
if strings.HasPrefix(match[1], "-")
```

It's a bit more obvious what's happening.

Reply:
Let me know if it still makes sense for you, i was able to put it in one line with `strings.CutPrefix`